packages/hash/api/src/model/entity.model.ts

```diff
@@ -633,26 +633,22 @@ class __Entity {
         });
       }
 
-      // insert links between entities in tree
-      // No transactions, so can run concurrently
-      await Promise.all(
-        entities.map(async ({ link, entity }) => {
-          if (link) {
-            const parentEntity = entities[link.parentIndex];
-            if (!parentEntity) {
-              throw new ApolloError("Could not find parent entity");
-            }
-            // links are created as an outgoing link from the parent entity to the children.
-            return await parentEntity.entity.createOutgoingLink(client, {
-              createdByAccountId: user.accountId,
-              destination: entity,
-              stringifiedPath: link.meta.path,
-              index: link.meta.index ?? undefined,
-            });
+      /** @todo: implement and use method for creating these links concurrently */
+      for (const { link, entity } of entities) {
+        if (link) {
+          const parentEntity = entities[link.parentIndex];
+          if (!parentEntity) {
+            throw new ApolloError("Could not find parent entity");
           }
-          return null;
-        }),
-      );
+          // links are created as an outgoing link from the parent entity to the children.
+          await parentEntity.entity.createOutgoingLink(client, {
+            createdByAccountId: user.accountId,
+            destination: entity,
+            stringifiedPath: link.meta.path,
+            index: link.meta.index ?? undefined,
+          });
+        }
+      }
```

Comment:
@thehabbos007 unfortunately at the moment we cannot call multiple `createOutgoingLink ` concurrently in the same transaction, apologies for having missed this in the review of your PR.

I've created [a task](https://app.asana.com/0/1201707629991380/1201817757879399/f) to find an approach that would let us do this.

Reply:
Good spot, thanks for changing it out for now. We have an [old task](https://app.asana.com/0/1201095311341924/1201646107470464/f) that touches on some of same underlying concurrent transaction issues. These two tasks might be good to group